LLaMA_LoRA.ipynb

```diff
@@ -60,20 +60,15 @@
         "# @title A small workaround { display-mode: \"form\" }\n",
         "# @markdown Don't panic if you see an error here. Just click the `RESTART RUNTIME` button in the output below, then Run All again.\n",
         "# @markdown The error will disappear on the next run.\n",
-        "!pip install Pillow==9.3.0 numpy==1.23.5\n",
+        "%pip install Pillow==9.3.0 numpy==1.23.5\n",
         "\n",
+        "import pkg_resources as r\n",
         "import PIL\n",
-        "major, minor = map(float, PIL.__version__.split(\".\")[:2])\n",
-        "version_float = major + minor / 10**len(str(minor))\n",
-        "print('PIL', version_float)\n",
-        "if version_float < 9.003:\n",
-        "    raise Exception(\"Restart the runtime by clicking the 'RESTART RUNTIME' button above (or Runtime > Restart Runtime).\")\n",
-        "\n",
         "import numpy\n",
-        "major, minor = map(float, numpy.__version__.split(\".\")[:2])\n",
-        "version_float = major + minor / 10**len(str(minor))\n",
-        "print('numpy', version_float)\n",
-        "if version_float < 1.0023:\n",
+        "for module, min_version in [(PIL, \"9.3\"), (numpy, \"1.23\")]:\n",
+        "  lib_version = r.parse_version(module.__version__)\n",
+        "  print(module.__name__, lib_version)\n",
+        "  if lib_version < r.parse_version(min_version):\n",
         "    raise Exception(\"Restart the runtime by clicking the 'RESTART RUNTIME' button above (or Runtime > Restart Runtime).\")"
```

Comment:
could use `exit()` as well which would "restart" it automatically, but colab doesn't seem to like it

Reply:
This reminds me that I've been using `os.kill(os.getpid(), 9)` to restart Colab notebooks before. It actually works quite well, as only the IPython will be restarted, with all the files remaining. Just felt it was too violent to be used here...